catapult/deploy.py

```diff
@@ -65,9 +65,13 @@ def start(
 
     last_deploy = next(get_releases(client, name, bucket=bucket), None)
 
-    releases = list(
-        get_releases(client, name, since=last_deploy.version if last_deploy else 0)
-    )
+    last_deployed_version = int(last_deploy.version) if last_deploy else 0
+    if version is not None:
+        since = min(int(version), last_deployed_version)
+    else:
+        since = last_deployed_version
+
+    releases = list(get_releases(client, name, since=since))
```

Comment:
The key problem was that we assumed that `last_deploy` was the earlier of the two, which then meant that `commits` (passed into `keep_only_commits`) would not contain the majority of the commits to be rolled back.